pyo3-introspection/src/model.rs

```diff
@@ -14,4 +14,26 @@ pub struct Class {
 #[derive(Debug, Eq, PartialEq, Clone, Hash)]
 pub struct Function {
     pub name: String,
+    pub arguments: Arguments,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Hash)]
+pub struct Arguments {
```

Comment:
I am mostly happy with this, though I do somewhat wish that this encoded the fact that vararg and kwarg cannot be defaulted (and also that required positional arguments cannot follow positional arguments with defaults).

In practice this is probably good enough, as long as we add suitable checks elsewhere.

Reply:
I have added a new `VariableLengthArgument` type without the `default` field.